qiskit_superstaq/custom_gates.py

```diff
@@ -51,13 +60,25 @@ def __array__(self, dtype: Type = None) -> np.ndarray:
             dtype=dtype,
         )
 
+        # sandwiched rx gate commutes and can just be multiplied with non-sandwiched part:
+        return mat @ np.kron(
+            np.asarray(qiskit.circuit.library.RXGate(self.sandwich_rx_rads), dtype=dtype),
+            np.eye(2, dtype=dtype),
+        )
+
     def __repr__(self) -> str:
+        args = f"'{self.polarity}'"
+        if self.sandwich_rx_rads:
+            args += f", sandwich_rx_rads={self.sandwich_rx_rads}"
         if self.label:
-            return f"qiskit_superstaq.AceCR('{self.polarity}', label='{self.label}')"
-        return f"qiskit_superstaq.AceCR('{self.polarity}')"
+            args += f", label='{self.label}'"
+        return f"qiskit_superstaq.AceCR({args})"
 
     def __str__(self) -> str:
-        return f"AceCR{self.polarity}"
+        if not self.sandwich_rx_rads:
+            return f"AceCR{self.polarity}"
+        args = qiskit.circuit.tools.pi_check(self.sandwich_rx_rads, ndigits=8, output="qasm")
+        return f"AceCR{self.polarity}|RXGate({args})|"
```

Comment:
nits:
```suggestion
        arg = qiskit.circuit.tools.pi_check(self.sandwich_rx_rads, ndigits=8, output="qasm")
        return f"AceCR{self.polarity}|RXGate({arg})|"
```
or
```suggestion
        arg = qiskit.circuit.tools.pi_check(self.sandwich_rx_rads, ndigits=8, output="qasm")
        return f"AceCR{self.polarity}|Rx({arg})|"
```
(the latter looks cleaner to me but either is fine)

Reply:
(for the future) something I just noticed: none of the gates in `qiskit.circuit.library` define `__repr__` or `__str__` methods. For the versions of qiskit we currently support, these default to the python built-in definitions, e.g.:
```
>>> str(qiskit.circuit.library.RXGate(1))
'<qiskit.circuit.library.standard_gates.rx.RXGate object at 0xabc123abc123>'  # qiskit-terra 0.18.3
```
but in later versions they are inherited from `qiskit.circuit.Instruction`:
```
>>> str(qiskit.circuit.library.RXGate(1))
"Instruction(name='rx', num_qubits=1, num_clbits=0, params=[1])"  # qiskit-terra 0.19.2
```
which makes me wonder if we should eventually be removing `__repr__` and/or `__str__` for custom_gates so that we fall back on the same

Reply:
> removing `__repr__` and/or `__str__` for custom_gates so that we fall back on the same

I can see us doing this for `str` but their `repr` doesn't seem to do what we want.

```python
>>> import qiskit
>>> repr(qiskit.circuit.library.RXGate(1))
'<qiskit.circuit.library.standard_gates.rx.RXGate object at 0x7fba8f7fa640>'
>>> eval(repr(qiskit.circuit.library.RXGate(1)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 1
    <qiskit.circuit.library.standard_gates.rx.RXGate object at 0x7fba8f7fa640>
    ^
SyntaxError: invalid syntax
>>> 


```

Reply:
Hmm, their `str` doesn't tell us the parameter of that `RXGate`.

Reply:
yeah, I think both should stay as they are as long as we're using `qiskit-terra==0.18.*`. But later versions do ~kinda work:
```
>>> qiskit.__version__
'0.19.2'
>>> str(qiskit.circuit.library.RXGate(1))
"Instruction(name='rx', num_qubits=1, num_clbits=0, params=[1])"
>>> eval(repr(qiskit.circuit.library.RXGate(1)))
Instruction(name='rx', num_qubits=1, num_clbits=0, params=[1])
```
which seems ok for `str` but still isn't exactly what we want for `repr`